server/src/test/java/com/google/sps/AdRowProcessorTest.java

```diff
@@ -1,30 +1,186 @@
 package com.google.sps;
-
 import com.google.sps.utils.Ad;
 import com.google.sps.utils.AdRowProcessor;
-
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
-/*
- * 
- */
 @RunWith(JUnit4.class)
 public final class AdRowProcessorTest {
   @Test
-  public void equality() {
-    String[] row = {"CR53959591413","FRIENDS OF SUZY GLOWIAK","2018-09-22","2018-11-06","≤ 10k",
-            "Not targeted","Not targeted","Illinois","100","1000.0","The reasonable choice",
-            "Ad suzyforsenate.com/about","Suzy Glowiak is the common-sense pick for the 24th.",
-            "Headline Sentiment", "Headline Terms", "Content Sentiment", "Content Terms"};
+  public void Should_ParseInputWithoutError_When_InputFormatExpected() {
+    String[] row = {"Id","Advertiser","2000-01-01","2000-01-01","≤ 10k",
+            "Not targeted","Not targeted","Not targeted","0","0",
+            "Headline", "Ad Link","Content", "Headline Sentiment", "Headline Terms", 
+            "Content Sentiment", "Content Terms"};
     Ad ad = AdRowProcessor.convertRowToAd(row);
     
-    Assert.assertEquals(ad.getId(), "CR53959591413");
-    Assert.assertEquals(ad.getId(), "CR53959591413");
-   
+    Assert.assertEquals(ad.getId(), "Id");
+    Assert.assertEquals(ad.getAdvertiser(), "Advertiser");
+    Assert.assertEquals(ad.getStartDate(), "2000-01-01");
+    Assert.assertEquals(ad.getEndDate(), "2000-01-01");
+    Assert.assertEquals(ad.getImpressionsMin(), 0);
+    Assert.assertEquals(ad.getImpressionsMax(), 10000);
+    Assert.assertEquals(ad.getIsTargetingAge(), false);
+    Assert.assertEquals(ad.getGenderTargets(), Arrays.asList("Not targeted"));
+    Assert.assertEquals(ad.getGeoTargets(), Arrays.asList("Not targeted"));
+    Assert.assertEquals(ad.getSpendMin(), 0);
```

Comment:
I feel like these assertions will be repeated if you ever want to test all the ad properties again (feel like you probably will), so perhaps instead make a helper function for it and pass in the expected values as an array or something.